FirebaseAppDistribution/Sources/FIRAppDistribution.m

```diff
@@ -277,39 +278,23 @@ - (void)checkForUpdateWithCompletion:(FIRAppDistributionUpdateCheckCompletion)co
   if ([self isTesterSignedIn]) {
     [self fetchNewLatestRelease:completion];
   } else {
-    UIAlertController *alert = [UIAlertController
-        alertControllerWithTitle:@"Enable in-app alerts"
-                         message:@"Sign in with your Firebase App Distribution Google account to "
-                                 @"turn on in-app alerts for new test releases."
-                  preferredStyle:UIAlertControllerStyleAlert];
-
-    UIAlertAction *yesButton =
-        [UIAlertAction actionWithTitle:@"Turn on"
-                                 style:UIAlertActionStyleDefault
-                               handler:^(UIAlertAction *action) {
-                                 [self signInTesterWithCompletion:^(NSError *_Nullable error) {
-                                   if (error) {
-                                     completion(nil, error);
-                                     return;
-                                   }
-
-                                   [self fetchNewLatestRelease:completion];
-                                 }];
-                               }];
-
-    UIAlertAction *noButton = [UIAlertAction actionWithTitle:@"Not now"
-                                                       style:UIAlertActionStyleDefault
-                                                     handler:^(UIAlertAction *action) {
-                                                       // precaution to ensure window gets destroyed
-                                                       [self.UIService resetUIState];
-                                                       completion(nil, nil);
-                                                     }];
-
-    [alert addAction:noButton];
-    [alert addAction:yesButton];
-
-    // Create an empty window + viewController to host the Safari UI.
-    [self.UIService showUIAlert:alert];
```

Comment:
Since we don't actually use the UIAlertAction we could simplify this `showUIAlert` to have a single completion block with a `userCancelled` boolean - Something like this - 
```
- (void)showUIAlertWithCompletion:(void (^)(BOOL userCancelled))completion {
  UIAlertController *alert = [UIAlertController
      alertControllerWithTitle:@"Enable in-app alerts"
                       message:@"Sign in with your Firebase App Distribution Google account to "
                               @"turn on in-app alerts for new test releases."
                preferredStyle:UIAlertControllerStyleAlert];

  UIAlertAction *yesButton = [UIAlertAction actionWithTitle:@"Turn on"
                                                      style:UIAlertActionStyleDefault
                                                    handler:^(UIAlertAction *action) {
                                                        completion(false);
                                                    }];

  UIAlertAction *noButton = [UIAlertAction actionWithTitle:@"Not now"
                                                     style:UIAlertActionStyleDefault
                                                   handler:^(UIAlertAction *action) {
                                                     [self resetUIState];
                                                     completion(true);
                                                   }];

  [alert addAction:noButton];
  [alert addAction:yesButton];

  // Create an empty window + viewController to host the Safari UI.
  [self showUIAlert:alert];
}
```

You can call this method like this - 
```
    
    [self showUIAlertWithCompletion:^(BOOL userCancelled) {
        if(!userCancelled) {
            [self signInTesterWithCompletion:^(NSError *_Nullable error) {
              if (error) {
                completion(nil, error);
                return;
              }

              [self fetchNewLatestRelease:completion];
            }];
        } else {
            completion(nil, nil);
        }
    }];
```

Reply:
ooo, yeah, I like that a lot!